tutor/hooks/catalog.py

```diff
@@ -257,6 +257,14 @@ def your_filter_callback(some_data):
     #:   names must be prefixed with the plugin name in all-caps.
     CONFIG_UNIQUE: Filter[list[tuple[str, Any]], []] = Filter()
 
+    #: Declare unique configuration settings that must be saved in the user ``config.yml`` file. This is where
```

Comment:
The docstring should be updated. Please try to emphasise that most people should not need this, and describe in which specific scenarios this filter should be used.

Reply:
Updated to for extra specificity

```
    #: Used to declare unique key:value pairs in the ``config.yml`` file that will be overwritten on ``tutor config save``.
    #: This is where you should declare passwords and other secrets that need to be fetched live from an external secrets
    #: store. Most users will not need to use this filter but it will allow you to programmatically fetch and set secrets
    #: from an external secrets store such as AWS Secrets Manager via boto3.
    #:
    #: Values passed in to this filter will overwrite existing values in the ``config.yml`` file.
    #:
    #: :parameter list[tuple[str, ...]] items: list of (name, value) new settings. All
    #:   names must be prefixed with the plugin name in all-caps.
    CONFIG_USER: Filter[list[tuple[str, Any]], []] = Filter()
```